### PR TITLE
fix(config): respect useOldToken config as default for all public methods

### LINKED ISSUE #96: UseOldToken needed to be passed
Even though I configure to use old token by:
```ScAuthenticationServiceProvider.useOldToken(true);```

I have to pass the flag in the methods anyway for most of public methods, like ```login, silentLogin, getToken, setToken``` etc.

### PR DESCRIPTION
**Pull request by cjhowald**
Fixes #96 

All public methods now default to the `useOldToken` configuration if that parameter is not defined.

### CODE REVIEW

File: src/authentication-service.js [hunk 1]
```diff
@@ -572,7 +572,7 @@
      * @param redirectUrl URL to redirect to after successfully login in. If not present, no redirection is performed
      */
     function clearCredentialsAndRedirectToLogin(redirectUrl) {
-      service.clearCredentials(self.shouldUseOldToken());
+      service.clearCredentials();
```
Comment (dami-gg) [inline src/authentication-service.js]: Is this going to be called always with the new token?

File: src/authentication-service.js [hunk 2]
```diff
@@ -251,9 +251,9 @@
      * Removes the user credentials from the storage and logs the user out without redirecting anywhere
      */
     function silentLogout() {
-      return invalidateToken(getToken(self.shouldUseOldToken()))
+      return invalidateToken(getToken())
         .then(function () {
-          return invalidateSessionToken(getSessionToken(self.shouldUseOldToken()));
+          return invalidateSessionToken(getSessionToken());
```
Comment (dami-gg) [inline src/authentication-service.js]: Much cleaner now. Good refactoring

File: src/authentication-service.js (hunk 1 above)
Comment (cjhowald) [inline src/authentication-service.js]: `clearCredentials` like the rest of the public APIs defaults to the `useOldToken` configuration.

**Comment by dami-gg** [comment]
:+1:

**Comment by cjhowald** [comment]
👍

**Pull request merged**